android/wolfssljni-ndk-gradle/app/src/main/java/com/wolfssl/wolfssljni_ndk_gradle/MainActivity.java

```diff
@@ -20,111 +22,180 @@ public class MainActivity extends AppCompatActivity {
+    };
 
-                        sock.startHandshake();
+    private final View.OnClickListener wcBenchmarkListener = v -> {
+        setDisplayText("Running wolfCrypt benchmark.\n" +
+                "See logcat output for details (adb logcat).\n" +
+                "...\n");
+
+        Executors.newSingleThreadExecutor().execute(() -> {
+            int ret = benchWolfCrypt();
+            if (ret == 0) {
+                appendDisplayText("BENCHMARK PASSED\n");
+            } else {
+                appendDisplayText("BENCHMARK FAILED!\nInspect logcat.");
+            }
+            appendDisplayText("\nTo adjust wolfSSL configuration, please edit\n" +
+                      "add_definitions() in CMakeLists.txt.");
+        });
+    };
 
-                        sock.close();
+    private final View.OnClickListener sslSocketConnectListener = v -> {
+        setDisplayText("Making simple SSL/TLS connection to:\n" +
+                host + ":" + port + "\n" +
+                "See logcat output for details (adb logcat).\n" +
+                "...\n");
+        Executors.newSingleThreadExecutor().execute(() -> {
+            try {
+                /* Enable wolfJSSE debug messages */
+                System.setProperty("wolfjsse.debug", "true");
+
+                long[] ts = new long[10];
+
+                ts[0] = System.currentTimeMillis();
+                WolfSSLProvider wolfProv = new WolfSSLProvider();
+                appendDisplayText("A. Inserting Provider " + wolfProv.getName() + "\n");
+                Security.insertProviderAt(wolfProv, 1);
+
+                ts[1] = System.currentTimeMillis();
+                connectCount++;
+                /* Unique devId's are compiled into wolfssljni */
+                int devId = 0x56490420;
+                if ((connectCount % 2) == 0) {
+                    appendDisplayText("B. Using hardware offload\n");
+                    wolfProv.registerDevId(devId);
+                    wolfProv.setDevId(devId);
+                } else {
+                    appendDisplayText("B. Using software crypto.\n");
+                    wolfProv.setDevId(WolfSSL.INVALID_DEVID);
+                }
 
-                        tv.append("TLS Connection Successful!");
+                ts[2] = System.currentTimeMillis();
+                appendDisplayText("C. Setting up context and socket\n");
+
+                /* not setting up KeyStore or TrustStore, wolfJSSE will load
+                 * CA certs from the Android system KeyStore by default. */
+                SSLContext ctx = SSLContext.getInstance("TLS", wolfProv.getName());
+                ctx.init(null, null, null);
+
+                SSLSocketFactory sf = ctx.getSocketFactory();
+                SSLSocket sock = (SSLSocket) sf.createSocket(host, port);
+
+                /* Limit cipherSuites and protocol */
+                String[] cipherSuites = new String[] {"TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA"};
```

Comment:
Generally for this sample app, I don't think it makes sense to limit the cipher suite and protocol to a single value.  Maybe this logic can be gated around a specific variable being set?